src/prommis/leaching/leach_train.py

```diff
@@ -0,0 +1,315 @@
+
+    CONFIG.declare(
+        "ssc_solver_options",
+        ConfigDict(
+            implicit=True,
+            description="Dict of arguments for solver calls by ssc_solver",
+        ),
+    )
+    CONFIG.declare(
+        "calculate_variable_options",
+        ConfigDict(
+            implicit=True,
+            description="Dict of options to pass to 1x1 block solver",
+            doc="Dict of options to pass to calc_var_kwds argument in "
+            "scc_solver method.",
+        ),
+    )
+
+    def initialization_routine(
+        self,
+        model: Block,
+    ):
+        """
+        Initialization routine for MSContactor Blocks.
+
+        Args:
+            model: model to be initialized
+
+        Returns:
+            None
+        """
+        # Initialize MSContactor
+        msc_init = model.mscontactor.default_initializer(
+            ssc_solver_options=self.config.ssc_solver_options,
+            calculate_variable_options=self.config.calculate_variable_options,
+        )
+        return msc_init.initialize(model.mscontactor)
+
+
+StreamCONFIG = ConfigDict()
+StreamCONFIG.declare(
+    "property_package",
+    ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description="Property package to use for given stream",
+        doc="""Property parameter object used to define property calculations for given stream,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+    ),
+)
+StreamCONFIG.declare(
+    "property_package_args",
+    ConfigDict(
+        implicit=True,
+        description="Dict of arguments to use for constructing property package",
+        doc="""A ConfigDict with arguments to be passed to property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}""",
+    ),
+)
+StreamCONFIG.declare(
+    "has_energy_balance",
+    ConfigValue(
+        default=False,
+        domain=Bool,
+        doc="Bool indicating whether to include energy balance for stream. Default=True.",
+    ),
+)
+StreamCONFIG.declare(
+    "has_pressure_balance",
+    ConfigValue(
+        default=True,
+        domain=Bool,
+        doc="Bool indicating whether to include pressure balance for stream. Default=True.",
+    ),
+)
+
+
+@declare_process_block_class("LeachingTrain")
+class LeachingTrainData(UnitModelBlockData):
+    """
+    Leaching Train Unit Model Class
+    """
+
+    # Set default initializer
+    default_initializer = LeachingTrainInitializer
+
+    CONFIG = UnitModelBlockData.CONFIG()
+
+    CONFIG.declare(
+        "liquid_phase",
+        StreamCONFIG(
+            description="Liquid phase properties",
+        ),
+    )
+    CONFIG.declare(
+        "solid_phase",
+        StreamCONFIG(
+            description="Solid stream properties",
```

Comment:
```suggestion
            description="Solid phase properties",
```
A bit nitpicky, but is there a reason the word "stream" was used here over "phase". Either way, I feel like the descriptions for the liquid and solid phase should use the same verbiage.